healsparse/visu_func.py

```diff
@@ -0,0 +1,218 @@
+        yticks = np.linspace(-89.99999, 89.99999, ny).tolist()
+        # This shows the full globe
+        ax.set_global()
+    # Get the vectors that point to the position of the corners, reshape to feed into vec2ang to get the position
+    # in ra, dec
+    corner_ra, corner_dec = hp.vec2ang(np.transpose(hp.boundaries(HealSparseMap.nsideSparse, HealSparseMap.validPixels,
+                                                                  nest=True), axes=[0,2,1]).reshape(-1,3), lonlat=True)
+    # Now reshape to create the corresponding matplotlib polygons:
+    uv_verts = np.array([corner_ra.reshape(-1, 4),
+                     corner_dec.reshape(-1, 4)]).transpose(1, 2, 0)
+    
+    # Get the colormap
+    cmap = plt.get_cmap(cmap_name)
+    data= HealSparseMap.getValuePixel(HealSparseMap.validPixels)
+    # Normalize the colormap
+    if vmin is None:
+        vmin = np.min(data[data!=HealSparseMap._sentinel])
+    if vmax is None:
+        vmax = np.max(data[data!=HealSparseMap._sentinel])
+    norm = plt.Normalize(vmin, vmax)
+    # Create the polygons and transform them with the Geodetic CRS to get them projected in the map without wrapping
+    polycoll = PolyCollection(uv_verts,edgecolor='none',array=data,
+                              norm=norm, transform=cartopy.crs.Geodetic())
+    polycoll.set_cmap(cmap)
+    fig.colorbar(polycoll, ax=ax, orientation='vertical', label=colorlabel)
+    # Add the PolyCollection to the axes
+    ax.add_collection(polycoll, autolim=False)
+    if extent is not None:
+        ax.set_extent(extent, crs=cartopy.crs.PlateCarree())
+    # Add the grid lines and labels
+    if projection in ['PlateCarree','Mercator']:
+        # This draws the labels in the ticks for these projections
+        ax.gridlines(xlocs=xticks, ylocs=yticks, draw_labels=True)
+    else:
+        # The ticks are not implemented for other projections so we prepare a workaround
+        # This is needed in order to add the ticks.
+        try:
+            fig.canvas.draw()
+            ax.gridlines(xlocs=xticks,ylocs=yticks)
+            ax.xaxis.set_major_formatter(LONGITUDE_FORMATTER) 
+            ax.yaxis.set_major_formatter(LATITUDE_FORMATTER)
+            custom_xticks(ax, xticks)
+            custom_yticks(ax, yticks)
+        except ImportError:
+            print('Shapely not found, ticks only implemented for Mercator and PlateCarree projections')
+        
+    ax.set_xlabel('RA [deg]', fontsize=16)
+    ax.set_ylabel('Dec [deg]', fontsize=16)
+    if show_coverage:
+        # Note: We can generalize this to other projections if needed but, for now, we'll stick to mollview
+        plt.axes(ax_cov)
+        hp.mollview(HealSparseMap.coverageMask, title='Coverage Mask', hold=True) #coverageMask is a healpix map
+        
+    if title is not None:
+        fig.suptitle(title, fontsize=20, ha='center')
+    if savename is not None:
+        fig.savefig(savename) 
+    fig.show()    
```

Comment:
need to make `show()` optional, so it won't crash when there is no display

Reply:
Also, I've got cartopy/pyproj installed but nothing shows on the screen, and nothing gets saved to a file